examples/errorhandlerbot.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This program is dedicated to the public domain under the CC0 license.
+
+"""
+This is a very simple example on how one could implement a custom error handler
+"""
+import html
+import json
+import logging
+import traceback
+
+from telegram import Update, ParseMode
+from telegram.ext import Updater, CallbackContext, CommandHandler
+
+logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+                    level=logging.INFO)
+
+logger = logging.getLogger(__name__)
+
+
+def error_handler(update: Update, context: CallbackContext):
+    """Log the error and send a telegram message with error details."""
+    # Log the error before we do anything else, so we can see it even if something breaks.
+    logger.error(msg="Exception while handling an update:", exc_info=context.error)
+
+    # traceback.format_exception returns the usual python message about an exception, but as a
+    # list of strings rather than a single string, so we have to join them together.
+    tb_list = traceback.format_exception(None, context.error, context.error.__traceback__)
+    tb = ''.join(tb_list)
+
+    # Build the message with some markup and additional information about what happened.
+    # You might need to add some logic to deal with messages longer than the 4096 character limit.
+    message = (
+        'An exception was raised while handling an update\n'
+        '<pre>update = {}</pre>\n\n'
+        '<pre>context.chat_data = {}</pre>\n\n'
+        '<pre>context.user_data = {}</pre>\n\n'
+        '<pre>{}</pre>'
+    ).format(
+        html.escape(json.dumps(update.to_dict(), indent=2, ensure_ascii=False)),
+        html.escape(str(context.chat_data)),
+        html.escape(str(context.user_data)),
+        html.escape(tb)
+    )
+
+    # Finally, send the message. You can replace this with something like
+    # context.bot.send_message(chat_id=DEVELOPER_CHAT_ID, text=message, parse_mode=ParseMode.HTML)
+    # to send the message to the bot developer rather than the user.
+    update.effective_message.reply_text(text=message, parse_mode=ParseMode.HTML)
```

Comment:
I just don't like that this example requires two edits 40 lines apart in order to run, and if you forget the first one things will break only after you use a command instead of immediately.

Would you rather
1) Leave it as it was, perhaps replacing Poolitzer's chat id with 123456789.
1) Change developer chat line to raise a an error if not modified e.g. to `DEVELOPER_CHAT_ID = ` (unfinished line) or `DEVELOPER_CHAT_ID = 0123456789` (leading zero is a syntax error in python 3)
1) Make bot token a module level constant so all the edits are in one place. I prefer this option.
1) Leave things as they are now, with assumption that the user will change the line to suit their needs.
1) Change the code to use update.effective_user to answer most updates and ignore the ones it can't.

Reply:
3 it is, then